### ISSUE TITLE
LinkML validation: Better handling of empty row errors

### ISSUE THREAD

**Issue by Shalsh23** [linkml/linkml#1403]
**Is your feature request related to a problem? Please describe.**
On validating data files using a linkml schema, if there are empty rows in the data file, it tries to validate them for each field and throws field specific  errors instead of recognizing that those are just empty rows. For example, this is a sample of errors received for the empty rows in the data files. These errors get resolved on pruning empty rows.

```
flmd_schema_path = 'flmd_linkml.yaml'
flmd_file_path = 'enzymes_dataProcessed_flmd.csv'

linkml-validate -s $flmd_schema_path -S flmd_files $flmd_file_path
```
Input files:
Schema: `flmd_linkml.yaml`
```
id: https://w3id.org/linkml/essdive/flmd
name: flmd_schema
description: schema for file level metadata (flmd) reporting format
prefixes:
  linkml: https://w3id.org/linkml/
imports:
  - linkml:types
default_range: string

classes:
  FileLevelMetadata:
    description: file level metadata for a data package uploaded on essdive
#    class_uri:

    attributes:

      File_Name:
        required: true
        description: Name of the file. 
          Provide the name of the associated file. File names should be unique and be as descriptive as possible about the file contents. Use only letters (e.g. CamelCase), numbers, and underscores. Do not include spaces. Hyphens allowed but not preferred. Use "*" wildcard when the FLMD applies to multiple files. For example - the same FLMD applies to all soil core files in this data package - "soil_cores_*.csv"


      File_Description:
        description: A brief description of the file.
          A brief description (minimum 30 characters) of the file and what distinguishes this file from other files in the data package. Include information about the type of data (images, observations, experimental, etc.)
        required: true

      Standard:
        description: Standard applied to the data file.
          Identify if an ESS-DIVE Reporting Format or any other data or metadata standard was applied to the data file.
        recommended: true

      UTC_Offset:
        description: Local Standard Time Offset.
          Report the Local Standard Time offset (+/- `hours`) or time zone (abbreviations allowed). Do not report time using Daylight Savings Time.
        recommended: true

      File_Version:
        description: The version of the file.
          This is the version of the data file being described in the FLMD. The data file version is assigned by the data provider and not by the system. This would change if the data file is updated after the data package is published. Changes should be explained in the Notes field.

      Contact:
        description: Contact for the file.
          Provide first and last name of the contact for the file. This should be someone with information already recorded at the Data Package Level.

      Date_Start:
        description: The earliest date in the file.
          All dates must follow the ISO 8601 standard. This field can be date only following the ISO 8601 standard (YYYY-MM-DD) and completed to known precision (e.g. YYYY-MM, YYYY). The FLMD Extractor can extract the start/end date from a CSV file following the CSV Reporting Format.
        range: date

      Date_End:
        description: The latest date in the file.
          All dates must follow the ISO 8601 standard. This field can be date only following the ISO 8601 standard (YYYY-MM-DD) and completed to known precision (e.g. YYYY-MM, YYYY). The FLMD Extractor can extract the start/end date from a CSV file following the CSV Reporting Format.
        range: date

      Northwest_Latitude_Coordinate:
        description: Northwest Latitude Coordinate for non-point location.
          This is part of a coordinate pair to form the Northwest corner of a bounding box for non-point locations. All geographic coordinates must be provided in WGS84 decimal format. It is strongly recommended to record the geographic coordinates in the FLMD in cases where the data file does not include geographic coordinates and the entire file consists of measurements collected at a single location. The FLMD Extractor can extract the geographic coordinates or bounding box coordinates from a CSV file following the CSV Reporting Format.
        range: double
        maximum_value: 90
        minimum_value: -90


      Northwest_Longitude_Coordinate:
        description: Northwest Longitude Coordinate for non-point location.
          This is part of a coordinate pair to form the Northwest corner of a bounding box for non-point locations. All geographic coordinates must be provided in WGS84 decimal format. It is strongly recommended to record the geographic coordinates in the FLMD in cases where the data file does not include geographic coordinates and the entire file consists of measurements collected at a single location. The FLMD Extractor can extract the geographic coordinates or bounding box coordinates from a CSV file following the CSV Reporting Format.
        range: double
        maximum_value: 180
        minimum_value: -180

      Southeast_Latitude_Coordinate:
        description: Southeast Latitude Coordinate for non-point location.
          This is part of a coordinate pair to form the Southeast corner of a bounding box for non-point locations. All geographic coordinates must be provided in WGS84 decimal format. It is strongly recommended to record the geographic coordinates in the FLMD in cases where the data file does not include geographic coordinates and the entire file consists of measurements collected at a single location. The FLMD Extractor can extract the geographic coordinates or bounding box coordinates from a CSV file following the CSV Reporting Format.
        range: double
        maximum_value: 90
        minimum_value: -90

      Southeast_Longitude_Coordinate:
        description: Southeast Longitude Coordinate for non-point location.
          This is part of a coordinate pair to form the Southeast corner of a bounding box for non-point locations. All geographic coordinates must be provided in WGS84 decimal format. It is strongly recommended to record the geographic coordinates in the FLMD in cases where the data file does not include geographic coordinates and the entire file consists of measurements collected at a single location. The FLMD Extractor can extract the geographic coordinates or bounding box coordinates from a CSV file following the CSV Reporting Format.
        range: double
        maximum_value: 180
        minimum_value: -180

      Latitude:
        description: Coordinate for single point location.
          This field is paired with "Longitude" to create a coordinate pair for a single point. All geographic coordinates must be provided in WGS84 decimal format. It is strongly recommended to record the geographic coordinates in the FLMD in cases where the data file does not include geographic coordinates and the entire file consists of measurements collected at a single location. The FLMD Extractor can extract the geographic coordinates or bounding box coordinates from a CSV file following the CSV Reporting Format.
        range: double
        maximum_value: 90
        minimum_value: -90

      Longitude:
        description: Coordinate for single point location.
          This field is paired with "Latitude" to create a coordinate pair for a single point. All geographic coordinates must be provided in WGS84 decimal format. It is strongly recommended to record the geographic coordinates in the FLMD in cases where the data file does not include geographic coordinates and the entire file consists of measurements collected at a single location. The FLMD Extractor can extract the geographic coordinates or bounding box coordinates from a CSV file following the CSV Reporting Format.
        range: double
        maximum_value: 180
        minimum_value: -180

      Missing_Value_Codes:
        description: Report missing value codes.
          Report all Missing Value Codes. For columns containing numeric data, use "-9999" as the missing value code (or modify to match significant figures given the data). For columns containing character data, use "N/A" as the missing value code. If entering more than one missing value code, use a vertical bar "|" or semicolon ";" instead of a comma or protect the comma with matching double quotation marks around the entire value. For more information about commas not meant to be a delimiter (e.g. used within a cell), refer to the Delimiter section of the CSV Reporting Format Detailed Guide.

      Data_Orientation:
        description: Orientation of tabular data.
          Describe how the data are organized within the data matrix. Choose between "horizontal" (i.e., data are organized in rows) or "vertical" (i.e., data are organized in columns).

      Notes:
        description: Any information the provider would like to add.
          Information provided would be data file specific. Details may include details on data file versioning, reporting format, software requirements, data quality, etc.

  Container:
    tree_root: true
    description: "list of flmd files"
    attributes:
      flmd_files:
        multivalued: true
        inlined_as_list: true
        range: FileLevelMetadata
```
Data file: `enzymes_dataProcessed_flmd.csv`
```
File_Name,File_Description,Standard,UTC_Offset,File_Version,Contact,Latitude,Longitude,Missing_Value_Codes,Data_Orientation
enzymeActivity_corr.csv,"""10,368"" corrected measurements of potential activity of the exoenzymes beta-glucosidase; leucine/leucyl aminopeptidase; and acid phosphatase in 18 soil samples collected from six depths in each of three replicate soil cores after incubation with each of eight substrate concentrations and under each of six temperatures. Values include quadruplicate measurements under each combination of conditions.",ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
enzymeActivity_corr_dd.csv,"CSV data dictionary for file ""enzymeActivity_corr.csv""",ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
enzymeKinetics_corr.csv,"313 corrected values of each of five Michaelis-Menten kinetic parameters of the exoenzymes beta-glucosidase; leucine/leucyl aminopeptidase; and acid phosphatase in 18 soil samples collected from six depths in each of three replicate soil cores; after incubation under each of six temperatures. Michaelis-Menten kinetics were modeled over quadruplicate measurements of enzyme activity under each of the cominations of conditions. Individual enzyme acivity measurements ar available in file ""enzymeActivity_corr.csv"". Parameters: maximum reaction velocity (Vmax; on dry soil and microbial biomass C basis); half-saturation/Michaelis constant (Km); and catalytic efficiency (CE; on dry soil and microbial biomass C basis).",ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
enzymeKinetics_corr_means.csv,"108 mean values and respective standard deviations and standard errors; of each of five Michaelis-Menten kinetic parameters of the exoenzymes beta-glucosidase; leucine/leucyl aminopeptidase; and acid phosphatase in triplicate soil samples collected from each of six soil depths; after incubation under each of six temperatures. Individual replicate values are available in file ""enzymeKinetics_corr.csv"". Michaelis-Menten kinetics were modeled over quadruplicate measurements of enzyme activity under each of the cominations of conditions (file ""enzymeActivity_corr.csv""). Parameters: maximum reaction velocity (Vmax; on dry soil and microbial biomass C basis); half-saturation/Michaelis constant (Km); and catalytic efficiency (CE; on dry soil and microbial biomass C basis).",ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
enzymeKinetics_corr_*_dd.csv,"CSV data dictionary for files ""enzymeKinetics_corr.csv"" and ""enzymeKinetics_corr_means.csv""",ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
enzymeTempSens_corr.csv,54 corrected values of each of multiple temperature sensitivity parameters of the exoenzymes beta-glucosidase; leucine/leucyl aminopeptidase; and acid phosphatase in 18 soil samples collected from six depths in each of three replicate soil cores. Parameters: activation energy and Q10 indices of Vmax; Km; and catalytic efficiency (linear Arrhenius model); temperature optimum; point of maximum temperature sensitivity; and change in heat capacity (non-linear Macromolecular Rate Theory model). Arrhenius parameters and Q10 were modeled over five temperatures between 4-35 degree Celsius and six temperatures between 4-50 degree Celsius; Macromolecular Rate Theory parameters were modeled over only six temperatures between 4-50 degree Celsius.,ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
enzymeTempSens_corr_means.csv,"18 mean values; and respective standard deviations and standard errors; of each of multiple temperature sensitivity parameters of the exoenzymes beta-glucosidase; leucine/leucyl aminopeptidase; and acid phosphatase in triplicate soil samples collected from each of six soil depths. Individual replicate values are available in file ""enzymeTempSens_corr.csv"". Parameters: activation energy and Q10 indices of Vmax; Km; and catalytic efficiency (linear Arrhenius model); temperature optimum; point of maximum temperature sensitivity; and change in heat capacity (non-linear Macromolecular Rate Theory model). Arrhenius parameters and Q10 were modeled over five temperatures between 4-35 degree Celsius and six temperatures between 4-50 degree Celsius; Macromolecular Rate Theory parameters were modeled over only six temperatures between 4-50 degree Celsius.",ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
enzymeTempSens_corr_*_dd.csv,"CSV data dictionary for files ""enzymeTempSens_corr.csv"" and ""enzymeTempSens_corr_means.csv""",ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
soilChemistry.csv,Soil and microbial biomass chemistry of 27 soil samples collected from nine depths in each of three replicate soil cores. Properties measured: GWC; DOC; TDN; MBC; MBN; and DOC:TDN and MBC:MBN ratios.,ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
soilChemistry_dd.csv,"CSV data dictionary for file ""soilChemistry.csv"".",ESS-DIVE Reporting Format for Comma-separated Values (CSV) File Structure 1.0.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
enzymes_sampleMetadata.csv,Sample registration spreadsheet for soil samples for Global Sample Numbers (IGSNs) through the System for Earth Sample Registration (SESAR). ,ESS-DIVE Sample ID and Metadata Reporting Format (IGSN-ESS) v1.1.0,- 5 hours,1,Ricardo Alves; Eoin Brodie,38.912,-120.661,-9999; N/A,horizontal
,,,,,,,,,
,,,,,,,,,
,,,,,,,,,
,,,,,,,,,
,,,,,,,,,
,,,,,,,,,
```

Output:
```
Loading from enzymes_dataProcessed_flmd2.tsv
Traceback (most recent call last):
  File "/Users/shalkishrivastava/shalkishrivastava_data/LBL/ess-dive/essdive-fusiondb/cep-tools/bin/linkml-validate", line 8, in <module>
    sys.exit(cli())
  File "/Users/shalkishrivastava/shalkishrivastava_data/LBL/ess-dive/essdive-fusiondb/cep-tools/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/shalkishrivastava/shalkishrivastava_data/LBL/ess-dive/essdive-fusiondb/cep-tools/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/shalkishrivastava/shalkishrivastava_data/LBL/ess-dive/essdive-fusiondb/cep-tools/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/shalkishrivastava/shalkishrivastava_data/LBL/ess-dive/essdive-fusiondb/cep-tools/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/shalkishrivastava/shalkishrivastava_data/LBL/ess-dive/essdive-fusiondb/cep-tools/lib/python3.10/site-packages/linkml/validators/jsonschemavalidator.py", line 143, in cli
    obj = loader.load(source=input, target_class=py_target_class, **inargs)
  File "/Users/shalkishrivastava/shalkishrivastava_data/LBL/ess-dive/essdive-fusiondb/cep-tools/lib/python3.10/site-packages/linkml_runtime/loaders/csv_loader.py", line 43, in load
    objs = unflatten_from_csv(source, config=config, **kwargs)
  File "/Users/shalkishrivastava/shalkishrivastava_data/LBL/ess-dive/essdive-fusiondb/cep-tools/lib/python3.10/site-packages/json_flattener/flattener.py", line 355, in unflatten_from_csv
    v = v.replace('\\n', '\n').replace('\\t', '\t')
AttributeError: 'list' object has no attribute 'replace'
```

We also tried using `linkml_validator` library and received similar errors due to empty rows.

Input: Same schema and data file as above

Code:
```
flmd_schema_path = "flmd_linkml.yaml"
csv_file_path = 'enzymes_dataProcessed_flmd.csv'

from pathlib import Path
import os
import pandas as pd

dir_path = Path(csv_file_path).parent
json_file_path = os.path.join(dir_path, 'ess_flmd.json')

# convert csv to json
# prune empty lines in csv file

 # this code works on dropping NAs
df = pd.DataFrame(pd.read_csv(csv_file_path, header=0, index_col=False)) #.dropna(axis='index', how='all')
df.to_json(json_file_path, orient="records")

from linkml_validator.validator import Validator

validator = Validator(schema=flmd_schema_path)
validator_itr = validator.validate_file(json_file_path, target_class="FileLevelMetadata")

for v in validator_itr:
    if(v.valid == False):
        print(v.validation_results[0].validation_messages)

os.remove(json_file_path)
```
Output:
```
SeverityEnum.error: 'Error'>, field='Longitude', value=None, message="None is not of type 'number'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='Missing_Value_Codes', value=None, message="None is not of type 'string'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='Standard', value=None, message="None is not of type 'string'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='UTC_Offset', value=None, message="None is not of type 'string'")]
[ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='Contact', value=None, message="None is not of type 'string'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='Data_Orientation', value=None, message="None is not of type 'string'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='File_Description', value=None, message="None is not of type 'string'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='File_Name', value=None, message="None is not of type 'string'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='File_Version', value=None, message="None is not of type 'number'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='Latitude', value=None, message="None is not of type 'number'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='Longitude', value=None, message="None is not of type 'number'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='Missing_Value_Codes', value=None, message="None is not of type 'string'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='Standard', value=None, message="None is not of type 'string'"), ValidationMessage(severity=<SeverityEnum.error: 'Error'>, field='UTC_Offset', value=None, message="None is not of type 'string'")]
```

**Describe the solution you'd like**
We would just like to give feedback or better error messages when empty rows lead to validation errors, perhaps even proceeding with the validation and just generating warnings for empty rows. 

**How important is this feature?** 
Medium

**When will use cases depending on this become relevant?** 
Mid-term

**Additional context**

**Comment by pkalita-lbl** [comment]
The is being addressed in the validation refactor (see #1494)

**Comment by pkalita-lbl** [comment]
As of LinkML 1.6.2, with the new validation framework, skipping empty rows in CSV files is now the default when using the `linkml-validate` CLI or the top-level `linkml.validator.validate_file` method. See [documentation](https://linkml.io/linkml/data/validating-data.html#the-linkml-validator-package-and-cli) for more information.
